interfaces/devdoc/threadpool_requirements.md

```diff
@@ -103,7 +103,7 @@ Note: `threadpool_close` will be deprecated and `threadpool_dispose` will perfor
 ### threadpool_create_work_item
 
 ```c
-MOCKABLE_FUNCTION(, THREADPOOL_WORK_ITEM_HANDLE, threadpool_create_work_item, THANDLE(THREADPOOL), threadpool, THREADPOOL_WORK_FUNCTION, work_function, void*, work_function_context);
+MOCKABLE_FUNCTION(, THANDLE(THREADPOOL_WORK_ITEM), threadpool_create_work_item, THANDLE(THREADPOOL), threadpool, THREADPOOL_WORK_FUNCTION, work_function, void*, work_function_context);
```

Comment:
>THANDLE(THREADPOOL), threadpool [](http://example.com/codeflow?start=80&length=31)

why does this function need to take `THANDLE(THREADPOOL), threadpool` as parameter?

At a later stage, we want to only be able to pass this newly created `THANDLE(THREADPOOL_WORK_ITEM)` to the original `THANDLE(THREADPOOL), threadpool` where it was created?

Or it can skip entirely the `THANDLE(THREADPOOL), threadpool` and allow scheduling the `THANDLE(THREADPOOL_WORK_ITEM)` on any existing threadpool?

Reply:
In Windows, the threadpool_ptr->tp_environment variable is saved and freed in dispose.